src/tools/r2rdump/R2RReader.cs

```diff
@@ -32,6 +32,116 @@ public enum Amd64Registers
         E15 = 15,
     }
 
+    /// <summary>
+    /// This structure represents a single precode fixup cell decoded from the
+    /// nibble-oriented per-method fixup blob. Each method entrypoint fixup
+    /// represents an array of cells that must be fixed up before the method
+    /// can start executing.
+    /// </summary>
+    public struct FixupCell
+    {
+        /// <summary>
+        /// Zero-based index of the import table within the import tables section.
+        /// </summary>
+        public uint TableIndex;
+
+        /// <summary>
+        /// Zero-based offset of the entry in the import table; it must be a multiple
+        /// of the target architecture pointer size.
+        /// </summary>
+        public uint CellOffset;
+
+        public FixupCell(uint tableIndex, uint cellOffset)
+        {
+            TableIndex = tableIndex;
+            CellOffset = cellOffset;
+        }
+    }
+
+    /// <summary>
+    /// Helper to read memory by 4-bit (half-byte) nibbles as is used for encoding
+    /// method fixups. More or less ported over from CoreCLR src\inc\nibblestream.h.
+    /// </summary>
+    class NibbleReader
```

Comment:
I think this should go into the NativeReader.cs file which contains other functions that read data or in its own file

Reply:
Thanks for the suggestion, fixed in 2nd commit. There is a counterpart NibbleWriter.cs in the R2R compiler.